compiler/main/driver.cpp

```diff
@@ -1670,14 +1671,16 @@ static void setGPUFlags() {
                " This may impact debuggability. To suppress this warning,"
                " compile with --no-checks explicitly");
 
-      fNoChecks = true;
-      fNoNilChecks    = true;
-      fNoBoundsChecks = true;
-      fNoFormalDomainChecks = true;
-      fNoLocalChecks  = true;
-      fNoStackChecks  = true;
-      fNoCastChecks = true;
-      fNoDivZeroChecks = true;
+       // if (doGpuCodegen()) {  we still can't do this.
```

Comment:
> I'm not sure I understand why you added this line and commented it out. Is it something you forgot to remove?

It was so that we don't forget discussing about it :)

Halfway through the implementation, I was somewhat hopeful that we can actually put the checks in with this mode so that you can have an easier debugging experience. But then it didn't work. I guess you can interpret this as something good -- that this mode doesn't do anything too differently compared to "real" gpu modes. By your comment, I would assume that you're in that camp. i.e., You wouldn't like this mode to do bounds checking even if we were able to?

Reply:
Yeah my inclination is for this mode to stay as close in behavior to what we have with other GPU modes as possible but your "these would be useful for an easier debugging" experience point seems compelling as well.

But since it didn't work anyway I guess we can avoid trying to think about what the behavior should be for now :-). Of course my hope would be that we could just make these checks part of "normal" GPU support soon.